scripts/staking_payouts.js

```diff
@@ -0,0 +1,367 @@
+  console.log("- Fetching eras info...");
+  const erasInfo = await fetchErasInfo(api);
+
+  const validators = pipe(
+    values,
+    pluck("validators"),
+    chain(keys),
+    (values) => new Set(values)
+  )(erasInfo.pointsByEra);
+
+  if (!validators.size) {
+    console.log("- Validator set is empty.");
+    return;
+  }
+
+  const erasToBePaid = await lastValueFrom(
+    getUnclaimedStashEras(api, erasInfo, validators).pipe(toArray())
+  );
+
+  const rewards = pipe(buildValidatorRewards, toPairs)(erasInfo, erasToBePaid);
+
+  if (isEmpty(rewards)) {
+    console.log("- No unpaid validator rewards found for recent eras.");
+    return;
+  }
+
+  console.log("- Payments need to be made:");
+
+  for (const [stashAddress, eras] of rewards) {
+    console.log(
+      ` * To \`${stashAddress}\`: ${eras.reduce(
+        ({ reward: a }, { reward: b }) => (a != null ? a.add(b) : b),
+        { reward: null }
+      )}`
+    );
+  }
+
+  const payoutTxs$ = from(rewards).pipe(
+    mergeMap(([stashAddress, eras]) =>
+      from(eras.map(assoc("stashAddress", stashAddress)))
+    ),
+    mapRx(({ stashAddress, era }) =>
+      api.tx.staking.payoutStakers(stashAddress, era)
+    ),
+    batchExtrinsics(api, BatchSize),
+    signAndSendPayouts(api, txSender)
+  );
+
+  await lastValueFrom(
+    payoutTxs$.pipe(
+      tap(({ hash, tx }) => {
+        console.log(` * Batch transaction finalized at block \`${hash}\`: [`);
+        for (const arg of tx.method.args[0]) {
+          console.log(
+            `   \`${arg.get("args").validator_stash}\` rewarded for era ${
+              arg.get("args").era
+            }`
+          );
+        }
+        console.log(" ]");
+      })
+    )
+  );
+}
+
+/**
+ * @typedef ErasInfo
+ * @prop {Array<BN>} eras
+ * @prop {Object<string, *>} pointsByEra
+ * @prop {Object<string, *>} rewardsByEra
+ * @prop {Object<string, *>} prefsByEra
+ */
+
+/**
+ * Fetches information for the recent eras with given limit.
+ *
+ * @param {*} api
+ * @returns {ErasInfo}
+ */
+const fetchErasInfo = async (api) => {
+  const eras = await api.derive.staking.erasHistoric();
+  const indexByEra = indexBy(o((era) => era.toString(), prop("era")));
+
+  const [pointsByEra, rewardsByEra, prefsByEra] = await Promise.all([
+    api.derive.staking._erasPoints(eras),
+    api.derive.staking._erasRewards(eras),
+    api.derive.staking._erasPrefs(eras),
+  ]).then(map(indexByEra));
+
+  return {
+    eras,
+    pointsByEra,
+    rewardsByEra,
+    prefsByEra,
+  };
+};
+
+/**
+ * @typedef StashEras
+ * @prop {Array<{ era: BN, eraReward: BN, reward: BN }>} eras
+ * @prop {string} stashAddress
+ */
+
+/**
+ * Retuns era rewards need to be paid grouped by stash ids.
+ *
+ * @param {*} api
+ * @param {ErasInfo} erasInfo
+ * @param {*} validators
+ * @returns {Array<StashEras>}
+ */
+const getUnclaimedStashEras = (api, { eras }, validators) => {
+  const singleStashEras = async (api, stashAddress, allEras) => {
```

Comment:
This is supposed to return unclaimed eras for the stash, correct? Can you name/document this function appropriately?